source/ranges.tex

```diff
@@ -525,7 +525,7 @@
   Otherwise, \tcode{(ranges::end(E) - ranges::begin(E))}
   if it is a valid expression and
   the types \tcode{I} and \tcode{S} of \tcode{ranges::begin(E)} and
-  \tcode{ranges::end(E)} model
+  \tcode{ranges::end(E)}, respectively, model
   \tcode{SizedSentinel<S, I>}\iref{iterator.concept.sizedsentinel} and
   \tcode{Forward\-Iterator<I>}.
```

Comment:
I'm worried that this will now be misread as `I` needing to model `SizedSentinel<S, I>` and `S` needing to model `ForwardIterator<I>`. How about also adding "both" after "model"? I also think the commas are too much of a disruption to the flow of the sentence, and "(respectively)" would read better.

Reply:
> How about also adding "both" after "model"?

Or combining the two constraint expressions into one:

Otherwise, `(ranges::end(E) - ranges::begin(E))` if it is a valid expression and the types `I` and `S` of `ranges::begin(E)` and `ranges::end(E)` (respectively) model `SizedSentinel<S, I> && ForwardIterator<I>`.

Reply:
Is "model `Thing1 && Thing2`" meaningful? I thought "model" means that some English side-condition is satisfied, and we don't have a side-condition specified for `Thing1 && Thing2` (though I suspect people will infer the right meaning anyway).

Reply:
Fixed by adding "both". I don't like the `&&`.